hls-graph/src/Development/IDE/Graph/Internal/Types.hs

```diff
@@ -5,7 +5,65 @@
 {-# LANGUAGE RecordWildCards    #-}
 {-# LANGUAGE ViewPatterns       #-}
 
-module Development.IDE.Graph.Internal.Types where
+module Development.IDE.Graph.Internal.Types
```

Comment:
Why am I adding explicit export list to this module?
I'm aware it's internal. But it exposes UnsafeMkKey + also things like newKey which use global reference to map with all keys.
If we make Key constructor opaque, it should be safe to add  `{-# COMPLETE Key #-}` pragma to get rid of "incomplete pattern match" warnings, because the only way to create new Key values are through newKey smart constructor.

Reply:
I don't have a strong opinion, although it is a rather long export list.

Reply:
I could try to make it shorter by splitting off KeyMap and KeySet into separate modules.. How does that sound?

Reply:
Or `Key` I guess, if that's the thing you want to be opaque. I don't want to derail you too much from your warning-fixing quest, but both of those sound like sensible refactorings to me.

Reply:
Ok, moved Key + KeyMap + KeySet stuff (because they depend on the internals) to separate module and removed explicit export list from Types module.